File: source/yeon/6주차/code/13904과제.cpp
// 백준 13904 과제
// cpp solved by Greedy algorithm + sort
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(void){
	int N, d, w, sum = 0;
	vector<pair<int,int>> pv;
	vector<int> result;
	scanf("%d",&N);
	for (int i = 0 ; i < N; i++){
		scanf("%d %d", &d, &w);
		pv.push_back(make_pair(d,w));
	}
	// range shold be 1-1000
	for(int i = 1000; i > 0; i--){
		for(int j = 0; j< N; j++){
			if(pv[j].first == i){
				result.push_back(pv[j].second);
			}
		}
		sort(result.begin(),result.end());
		if(!result.empty()){
			int max = result.back();
			sum += max;
			result.pop_back();
		}
	}
	printf("%d", sum);
	return 0;
}

File: source/yeon/6주차/code/13904과제.py
# 백준 13904 과제
# python solved by Greedy algorithm + sort

N = int(input())
arr = [list(map(int,input().split())) for _ in range(N)]
result = []
sum,ans = 0,0
# range should be from 1-1000
for i in range(1000,0,-1):
	for j in range(N):
		if arr[j][0] == i:
			result.append(arr[j][1])
	result.sort()
	if result :
		ans = result.pop()
		sum += ans

print(sum)